src/main/kotlin/com/angels/challenge/model/StoreSeasonPK.kt

```diff
@@ -0,0 +1,27 @@
+package com.angels.challenge.model
+
+import java.io.Serializable
+import java.util.*
+import javax.persistence.Column
+import javax.persistence.Embeddable
+
+@Embeddable
+class StoreSeasonPK : Serializable {
+    @Column(name = "store")
+    var store: String? = null
+
+    @Column(name = "season")
+    var season: String? = null
+
+    override fun hashCode(): Int {
+        return Objects.hash(store, season)
+    }
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
In kotlin is considered a bad practice to implement equals/hashcode functions manually.

Do you know what the correct approach would be?

Also, what brings you the need to have these methods in a persistence layer class?

Reply:
I was following a tutorial on how to do a composite PK for an entity and was running into some problems, then discovered that we should implements equals and hashcode in order for it to work properly. I don't know what is correct approach here, maybe the IDE provides an automatic tool to override these? 🤔

Reply:
Indeed, composite keys are a pain 😄 

Anyway, in Kotlin, there's the concept of [data class](https://kotlinlang.org/docs/data-classes.html) that gives you this kind of boilerplate for free, in order to avoid repeating the same code for every class.